src/Microsoft.CorrelationVector/CorrelationVector.cs

```diff
@@ -215,11 +215,56 @@ public string Value
         {
             get
             {
-                return string.Concat(this.baseVector, ".", this.extension,
+                return string.Concat(this.baseVector, ".", this.extension,
                     this.immutable ? CorrelationVector.TerminationSign : string.Empty);
             }
         }
 
+        /// <summary>
+        /// Gets an encoded vector base value from the given <see cref="Guid"/>.
+        /// </summary>
+        /// <param name="guid">The <see cref="Guid"/> to encode as a vector base.</param>
+        /// <returns>The encoded vector base value.</returns>
+        public static string GetBaseFromGuid(Guid guid)
+        {
+            byte[] bytes = guid.ToByteArray();
+
+            // Removes the base64 padding
+            return Convert.ToBase64String(bytes).Substring(0, CorrelationVector.BaseLengthV2);
+        }
+
+        /// <summary>
+        /// Gets the value of the correlation vector base encoded as a <see cref="Guid"/>.
+        /// </summary>
+        /// <returns>The <see cref="Guid"/> value of the encoded vector base.</returns>
+        public Guid GetBaseAsGuid()
```

Comment:
I don't mind separating out the functionality as I doubt all language implementations require this. Are you simply looking for a file separation (leveraging the partial nature of this class) or a true extension method (which would require making some private members internal)?

Reply:
If we were to keep the API across languages as close as possible, then a true extension method would probably be the way to go. Otherwise the file separation is fine.